circuits/src/main.nr

```diff
@@ -11,11 +11,38 @@ fn main(
   polynomial_commitment: Field,
   nullifier: pub [u8; 32],
 ) {
-  // Write your main logic here
+
+  let nullifier_hash = std::hash::blake2s(signature);
+  assert(nullifier_hash == nullifier);
+  std::println("checking nullifier passed");
+
+  let mut pub_key_x = [0; 32];
+  let mut pub_key_y = [0; 32];
+  for i in 0..32 {
+    pub_key_x[i] = pub_key[i];
+    pub_key_y[i] = pub_key[i + 32];
+  }
+  let signer = ecrecover::ecrecover(pub_key_x, pub_key_y, signature, hashed_message);
```

Comment:
You could have used this too - `let key = ecrecover::secp256k1::PubKey::from_unified(pub_key);` for conversion